catroid/src/main/java/org/catrobat/catroid/cloudmessaging/CloudMessaging.java

```diff
@@ -54,6 +69,50 @@ public void showNotification(CloudMessage cloudMessage) {
 		}
 	}
 
+	public static boolean initialize(Context context) {
+		if (isCloudNotificationsEnabled(context)) {
+			subscribeToFirebaseServices(context);
+			return true;
+		}
+		unsubscribeFromFirebaseServices(context);
+		return false;
+	}
+
+	@VisibleForTesting
+	public static void setIsCloudNotificationsEnabled(boolean isEnabled) {
+		isCloudNotificationsEnabled = isEnabled;
+	}
+
+	private static void subscribeToFirebaseServices(Context context) {
+		FirebaseMessaging.getInstance().subscribeToTopic(DEFAULT_TOPIC);
+		FirebaseMessaging.getInstance().subscribeToTopic(getLanguageCode(context));
+		FirebaseMessaging.getInstance().subscribeToTopic(getCountryCode(context));
+	}
+
+	private static void unsubscribeFromFirebaseServices(Context context) {
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(DEFAULT_TOPIC);
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(getLanguageCode(context));
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(getCountryCode(context));
+	}
+
+	private static boolean isCloudNotificationsEnabled(Context context) {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		return sharedPreferences.getBoolean(SettingsActivity.SETTINGS_CLOUD_NOTIFICATIONS, false) && isCloudNotificationsEnabled;
+	}
+
+	private static String getLanguageCode(Context context) {
+		String languageCode = context.getResources().getConfiguration().locale.getCountry();
```

Comment:
I haven't tested it but according to the docs, [` getDisplayLanguage()`](https://developer.android.com/reference/java/util/Locale.html#getDisplayLanguage()) is localized.

default DISPLAY locale is en_US => French
default DISPLAY locale is fr_FR => anglais

I'm not sure if there is a reliable way to get the standardized language code. However, we need a reliable way ^^

https://stackoverflow.com/a/8691091/2774399 looks interesting.

There is also a PR https://github.com/Catrobat/Catroid/pull/2332/files#diff-7a2a77ec325d68cbe50a2c6e57b0468aR53 already defining a list of all languages. However, it's far from final and maybe won't be merged that way.

The Firebase console also automatically provides languages. User properties can be used to check if a user should receive notifications. See
![image](https://user-images.githubusercontent.com/1794599/28827411-c2fb134c-76cd-11e7-8a9f-6571891ca344.png)
The problems, it can't be combined with topics and I don't know if it uses the app language or OS language :disappointed: There may be different type of notifications the user can subscribe to (events, news, hot projects, etc).